core/src/main/scala/kafka/server/ClientQuotaManager.scala

```diff
@@ -387,14 +360,15 @@ class ClientQuotaManager(private val config: ClientQuotaManagerConfig,
    * First sensor of the tuple is the quota enforcement sensor. Second one is the throttle time sensor
    */
   def getOrCreateQuotaSensors(sanitizedUser: String, clientId: String): ClientSensors = {
-    val clientQuotaEntity = quotaEntity(sanitizedUser, clientId)
+    val sanitizedClientId = Sanitizer.sanitize(clientId)
```

Comment:
Can we cache sanitized client-ids somewhere so that we dont have to sanitize for every request? Obviously they could change, but perhaps check and re-sanitize if required?

Reply:
I think we should also move `sanitizedUser` from `Session` to `KafkaChannel` perhaps since `Session` objects are now created for every request.

Reply:
@rajinisivaram I think the optimization of moving the sanitisedUser should be done in a separate PR, maybe for a point release.

Reply:
Currently once a connection is established , Sanitize is called once for the user and - with this PR - once per client-id. 

So I'd say about the suggested optimization of caching the sanitize(client-id) in CilentQuotaManager.getOrCreate... that as well should go in a separate PR.
At that time we could measure if caching is worth doing !

Reply:
@edoardocomar Hmm.. We used to sanitize user principal once per connection, but there was a recent PR (not this one) which changed it to once per request. That should be easy to fix though, but a separate PR is fine. For client-id caching, there is a bit more work involved since client-ids can change within a connection. Again I am ok with doing that later.